lib/elixir/lib/module/types/infer.ex

```diff
@@ -399,4 +514,54 @@ defmodule Module.Types.Infer do
 
   defp get_meta({_fun, meta, _args}) when is_list(meta), do: meta
   defp get_meta(_other), do: []
+
+  # TODO: Remove any non struct fields
+  #       Maybe not here...
```

Comment:
Let's expand this TODO? It needs to be a bit clearer what is the action point here or we should effectively tackle it. :)

Reply:
See the new comment here.